renovate.json

```diff
@@ -0,0 +1,35 @@
+{
+  "$schema": "https://docs.renovatebot.com/renovate-schema.json",
+  "extends": [
+    "config:base"
+  ],
+  "enabledManagers": ["npm"],
+  "rangeStrategy": "update-lockfile",
+  "schedule": [
```

Comment:
@rajsite this has been in draft for ages but I'm reviving it now. I'm starting with a conservative configuration:
1. opt out of updates for known vulnerabilities. We'll probably want to turn this on, but it adds a dozen or so new PRs we'll have to sort through so I'll wait until we're confident in this config before doing so.
2. opt out of updates to all dependencies initially except ones that are pretty trustworthy (NI and FAST). We'll want to add tanstack and d3 to this list soon. Other deps like storybook, angular, typescript, and build tooling are probably better to keep manual for a while.
3. limit PR creation to weekends only. I figure we can process them all on Monday mornings and bubble them up throughout the week, avoiding constant churn to our clients.

This will result in just a couple PRs being created (listed in PR description) for `fast-web-utilities` and `beachball`.

I'm open to feedback on all of the above.

Reply:
I revived this on an innovation day. Updated policies:
1. Still opting out of updates for known vulnerabilities to start
2. Still using an opt-in approach but I added configurations for several of our larger dependencies (D3, TanStack, tiptap/prosemirror, Playwright) since they've become important parts of our stack
3. Changed to monthly instead of weekly (see changefile discussion below for rationale)
4. Same `rangeStrategy` as above

You can see the PR description for a summary of what this achieves. It looks pretty sane to me. Some notes:
1. `@types/markdown-it` is in a separate PR from the rest of the Rich Text updates because it's a `major` bump. This seems sane to me: we probably want to give `major` bumps more scrutiny.
2. Not sure why the D3 section isn't picking up new versions of packages like `d3-array` ([3.2.4 is available](https://www.npmjs.com/package/d3-array) but we currently [depend on 3.0.4](https://github.com/ni/nimble/blob/4d32e3607dc2ddaf2cead3c9cd9d462d7e0d2c5e/packages/nimble-components/package.json#L78)).

Reply:
What if we just had two bump groups? Dependencies and devDependencies?

That's the approach I was taking when bumping manually. Would focus mainly on bumping dependencies and ignore devDepedencies 😅

Can't think of real specific value in bumping the dependencies all separately, just results in extra releases and work.

I like the two groups because I expect the dependencies upgrade to be much less noisy in package-lock than devDependencies. So nicer for inspecting changes resulting from dependencies stand-alone.

Reply:
![image](https://github.com/ni/nimble/assets/1588923/dcf6f1d1-99fa-48b8-9918-f918d8143fa0)

Sounds awful lol

Reply:
I see the update, but can we still do a controlled set of dependencies? at least to start?

Reply:
I think I've got it updated now. I'm good to try this approach.

Another downside of separate PRs: I was setting us/myself up for a painful cycle of one-by-one adding change files to each of those PRs, completing them, waiting for new packages to be published, and waiting for Renovate to update the PRs to resolve the package-lock.json conflicts.

One downside of batching PRs is that an issue with a single dependency would cause the PR to be blocked for other dependencies. But if that happens and we really want some of the updates, we can split the updates up manually. If it becomes common then we can re-evaluate this config.

Reply:
Know why it's trying to make the major version bump of `@​types/markdown-it` a separate PR from the other dependencies? Seems unnecessary..

Reply:
Looks like that's the default behavior: https://docs.renovatebot.com/faq/#renovates-default-behavior-for-majorminor-releases
And it can be disabled: https://docs.renovatebot.com/presets-default/#separatemajorreleases

Reply:
I am tempted to keep the default. Some of our dependencies have pretty disruptive major releases that require manual intervention to uptake (e.g. storybook, maybe FAST if they ever release their vNext). I think there'd be benefit to continuing to uptake minor/patch updates while we work through any major breaking changes.